Sources/SwiftDiagnostics/FixIt.swift

```diff
@@ -27,15 +27,45 @@ public protocol FixItMessage: Sendable {
   var fixItID: MessageID { get }
 }
 
+public protocol ReplacingChildData: Sendable {
+  associatedtype Parent: SyntaxProtocol
+  associatedtype Child: SyntaxProtocol
+
+  var replacementRange: Range<AbsolutePosition> { get }
+  var parent: Parent { get }
+  var newChild: Child { get }
+}
```

Comment:
Could you add comments to the protocol and its members? Also, I think it would be good to describe why this is a protocol and can’t be members on the `FixIt.Change.replaceChild` case.